scripts/main/contextMenu.js

```diff
@@ -317,36 +359,41 @@ contextMenu.photo = function (photoID, e) {
 	basicContext.show(items, e.originalEvent, contextMenu.close);
 };
 
+/**
+ * @param {string[]} photoIDs
+ * @returns {number}
+ */
 contextMenu.countSubAlbums = function (photoIDs) {
-	let count = 0;
+	if (album.json.albums === null || album.json.albums.length === 0) {
```

Comment:
This fails for smart and tag albums because they don't define the `albums` property. Either we need to fix this code or add the missing property (be it in the front end or on the server side).

Reply:
A quick solution can be found in: https://github.com/LycheeOrg/Lychee-front/pull/282/commits/e2eb24012f557a619c44fd04e8f5396ba72f5df1

I also renamed `photoIDs` into `albumEntryIDs`, because `photoIDs` did not make any sense.

Reply:
A proper solution can be found in: https://github.com/LycheeOrg/Lychee-front/pull/282/commits/4a1c25f33257aee409c71c22c4038a07bf94a5dc. The failing code can simply be deleted, because is unnecessary. The former name of the parameter `photoIDs` was entirely correct and ID of albums are never passed to this method. We don't need to deal with a mixed case here.

I was wondering why there has been an asymmetry between the context menu for multiple albums

https://github.com/LycheeOrg/Lychee-front/blob/8507d67c2818b20bc722a229041948865d54812e/scripts/main/contextMenu.js#L139-L140

and multiple photos

https://github.com/LycheeOrg/Lychee-front/blob/8507d67c2818b20bc722a229041948865d54812e/scripts/main/contextMenu.js#L339-L357

I asked myself why does the context menu for photos need to deal with a mixed case (and print an error) or call the context menu for albums if only albums IDs are passed while the context menu for albums may simply assume to only get invoked for album IDs? Indeed, it is completely unnecessary.

The event handlers for both context menu are symmetricly installed in

https://github.com/LycheeOrg/Lychee-front/blob/8507d67c2818b20bc722a229041948865d54812e/scripts/main/init.js#L273-L279

The object `multiselect` already counts the number of selected photo and albums IDs:

https://github.com/LycheeOrg/Lychee-front/blob/8507d67c2818b20bc722a229041948865d54812e/scripts/main/multiselect.js#L9-L14

and `addItem()` already prevents a mixed selection:

https://github.com/LycheeOrg/Lychee-front/blob/8507d67c2818b20bc722a229041948865d54812e/scripts/main/multiselect.js#L53-L65